packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -0,0 +1,244 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import * as KeyCode from 'keycode-js';
+import WorklistDataGridPropTypes from './proptypes/WorklistDataGridPropTypes';
+import styles from './WorklistDataGrid.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * String representing an attribute which identifies the element (or elements) that labels the grid.
+   */
+  ariaLabelledby: PropTypes.string,
+  /**
+   * String that labels the grid for accessibility.
+   */
+  ariaLabel: PropTypes.string,
+  /**
+   * String that will be used to identify the Grid. This value will be used as the id attribute of the overall grid container,
+   * and it will be used to prefix other id attributes used for internal componentry.
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Data for column headers. Columns will be presented in the order given.
+   */
+  columnHeaders: WorklistDataGridPropTypes.columnHeaderShape,
+  /**
+   * Data for content in the body of the Grid. Rows will be rendered in the order given.
+   */
+  rows: PropTypes.arrayOf(WorklistDataGridPropTypes.rowShape),
+  /**
+   * String that specifies the default column width. Any valid CSS width value is accepted.
+   * This value will be used if no overriding width value is provided on a per-column basis.
+   */
+  columnHeaderDefaultWidth: PropTypes.string,
+  /**
+   * String that specifies the default row height. Any valid CSS width value is accepted.
+   * This value will be used if no overriding height value is provided on a per-row basis.
+   */
+  rowDefaultHeight: PropTypes.string,
+  /**
+   * String that specifies the column height. Any valid CSS height value is accepted.
+   */
+  columnHeaderHeight: PropTypes.string,
+  /**
+   * Number indicating the index of the column that represents row header. Index is 0 based and cannot exceed one less than the number of columns in the grid.
+   */
+  rowHeaderIndex: PropTypes.number,
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+  columnHeaderDefaultWidth: '200px',
+  columnHeaderHeight: '2.5rem',
+  rowDefaultHeight: '2.5rem',
+};
+
+class WorklistDataGrid extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.focussedRow = 0;
+    this.focussedCol = 0;
+  }
+
+  componentDidMount() {
+    this.grid = document.querySelector(`#${this.props.id}`);
+
+    const focussedCell = this.grid.rows[this.focussedRow].cells[this.focussedCol];
+    focussedCell.setAttribute('tabIndex', '0');
+
+    // add event listeners to handle user key events in the grid.
+    this.grid.addEventListener('keydown', this.handleKeyDown);
+    this.grid.addEventListener('click', this.handleClick);
```

Comment:
This is not the proper way to add events in React.  You would use the onClick attribute in the JSX for the click event and onKeyDown for the keydown event.  We should not need to add JavaScript event listeners and be able to leverage the React synthetic events.  It might change the implementation of the event handlers.

Reply:
Modified to use OnClick & OnKeyDown